pr_testing/test_multiple_prs.sh

```diff
@@ -690,13 +690,32 @@ if [ "X$DO_STATIC_CHECKS" = "Xtrue" -a "X$CMSSW_PR" != X -a "$RUN_TESTS" = "true
   cp -R $WORKSPACE/$CMSSW_IB/llvm-analysis/*/* $WORKSPACE/llvm-analysis || true
   if $IS_DEV_BRANCH && [ $(grep ': error: ' $WORKSPACE/llvm-analysis/runStaticChecks.log | wc -l) -gt 0 ] ; then
     echo "EDM_ML_DEBUG_CHECKS;ERROR,Static Check build log,See Log,llvm-analysis/runStaticChecks.log" >> ${RESULTS_DIR}/static.txt
-  elif $IS_DEV_BRANCH && [ $(grep ': warning: ' $WORKSPACE/llvm-analysis/runStaticChecks.log | grep edm::eventsetup::EventSetupRecord::get | wc -l) -gt 0 ] ; then
-    echo "EDM_ML_DEBUG_CHECKS;WARNING,Static Check build log,See Log,llvm-analysis/runStaticChecks.log" >> ${RESULTS_DIR}/static.txt
+  elif $IS_DEV_BRANCH && [ $(grep ': warning: ' $WORKSPACE/llvm-analysis/runStaticChecks.log | grep '[threadsafety.ESRecordGetChecker]' | wc -l) -gt 0 ] ; then
+    grep ': warning: ' $WORKSPACE/llvm-analysis/runStaticChecks.log | grep edm::eventsetup::EventSetupRecord::get | sort -u > $WORKSPACE/llvm-analysis/esrget-sa.txt
+    echo "STATIC_CHECK_ESRGET;ERROR,Static analyzer EventSetupRecord::get warnings,See warnings log,llvm-analysis/esrget-sa.txt" >> ${RESULTS_DIR}/static.txt
   else
     echo "EDM_ML_DEBUG_CHECKS;OK,Static Check build log,See Log,llvm-analysis/runStaticChecks.log" >> ${RESULTS_DIR}/static.txt
   fi
   echo 'END OF STATIC CHECKS'
   echo '--------------------------------------'
+
+  echo 'CMS_CLANG_TIDY_CHECKS;OK,CMS clang-tidy checks output,See clang-tidy log,llvm-analysis/runCMSClangTidyChecks.log' >> ${RESULTS_DIR}/clangtidy.txt
+  echo '--------------------------------------'
+  echo "Changed files:"
+  echo ""
+  cat $WORKSPACE/changed-files
+  echo ""
+  USER_CXXFLAGS='-Wno-register -DEDM_ML_DEBUG -w' SCRAM_IGNORE_PACKAGES="Fireworks/% Utilities/StaticAnalyzers" USER_CODE_CHECKS='cms*' scram b -k -j ${NCPU2} code-checks USER_CODE_CHECKS_FILE="$WORKSPACE/changed-files" SCRAM_IGNORE_SUBDIRS=test 2>&1 | tee -a $WORKSPACE/llvm-analysis/runCMSClangTidyChecks.log
+  warncount=$(grep ': warning: ' $WORKSPACE/llvm-analysis/runCMSClangTidyChecks.log | grep 'call of function EventSetupRecord::get' | wc -l) 
+  if $IS_DEV_BRANCH && [ $warncount -gt 0 ]; then
+    grep ': warning: ' $WORKSPACE/llvm-analysis/runCMSClangTidyChecks.log | grep 'call of function EventSetupRecord::get'| sort -u > $WORKSPACE/llvm-analysis/cmsclangtidy.txt
+    echo "CMS_CLANG_TIDY_CHECKS_ESRGET;ERROR,CMS clang-tidy checks EventSetupRecord::get warnings,See clang-tidy warnings log,llvm-analysis/cmsclangtidy.txt" >> ${RESULTS_DIR}/clangtidy.txt
+    echo "**CMS Clang-Tidy warnings**: There are $warncount Clang-Tidy warnings. See ${PR_RESULT_URL}/llvm-analysis/cmsclangtidy.txt for details." >> ${RESULTS_DIR}/09-report.res
+  else
+    echo "CMS_CLANG_TIDY_CHECKS;OK,CMS clang-tidy checks output,See clang-tidy log,llvm-analysis/runCMSClangTidyChecks.log" >> ${RESULTS_DIR}/clangtidy.txt
+  fi
+  echo 'END OF CLANG-TIDY CHECKS'
+  echo '--------------------------------------'
```

Comment:
@gartung , one last comment ( I hope), shouldn;t this be run only for $IS_DEV_BRANCH ? I don;t think this will work for old release cycles .

Reply:
Correct. I will update.

Reply:
Line 710 has the check for IS_DEV_BRANCH already.